packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -13,88 +13,42 @@
 // limitations under the License.
 
 import * as _ from 'lodash';
+import DRange from 'drange';
 import { Trace, Span } from '../../../types/trace';
 import { ITableSpan } from './types';
 import colorGenerator from '../../../utils/color-generator';
 
 const serviceName = 'Service Name';
 const operationName = 'Operation Name';
 
-/**
- * Return the lowest startTime.
- */
-function getLowestStartTime(allOverlay: Span[]) {
-  let result;
-  const temp = _.minBy(allOverlay, function calc(a) {
-    return a.relativeStartTime;
-  });
-  if (temp !== undefined) {
-    result = { duration: temp.duration, lowestStartTime: temp.relativeStartTime };
-  } else {
-    result = { duration: -1, lowestStartTime: -1 };
-  }
-  return result;
-}
-
-/**
- * Determines whether the cut spans belong together and then calculates the duration.
- */
-function getDuration(lowestStartTime: number, duration: number, allOverlay: Span[]) {
-  let durationChange = duration;
-  let didDelete = false;
-  for (let i = 0; i < allOverlay.length; i++) {
-    if (lowestStartTime + durationChange >= allOverlay[i].relativeStartTime) {
-      if (lowestStartTime + durationChange < allOverlay[i].relativeStartTime + allOverlay[i].duration) {
-        const tempDuration =
-          allOverlay[i].relativeStartTime + allOverlay[i].duration - lowestStartTime + durationChange;
-        durationChange = tempDuration;
-      }
-      allOverlay.splice(i, 1);
-      didDelete = true;
-      break;
-    }
-  }
-  const result = { allOverlay, duration: durationChange, didDelete };
-  return result;
-}
-
-/**
- * Return the selfTime of overlay spans.
- */
-function onlyOverlay(allOverlay: Span[], allChildren: Span[], tempSelf: number, span: Span) {
-  let tempSelfChange = tempSelf;
-  let duration = 0;
-  let resultGetDuration = { allOverlay, duration, didDelete: false };
-  const noOverlay = _.difference(allChildren, allOverlay);
-  let lowestStartTime = 0;
-  let totalDuration = 0;
-  const result = getLowestStartTime(allOverlay);
-  lowestStartTime = result.lowestStartTime;
-  duration = result.duration;
-
-  do {
-    resultGetDuration = getDuration(lowestStartTime, duration, resultGetDuration.allOverlay);
-    if (!resultGetDuration.didDelete && resultGetDuration.allOverlay.length > 0) {
-      totalDuration = resultGetDuration.duration;
-      const temp = getLowestStartTime(resultGetDuration.allOverlay);
-      lowestStartTime = temp.lowestStartTime;
-      duration = temp.duration;
-    }
-  } while (resultGetDuration.allOverlay.length > 1);
-  duration = resultGetDuration.duration + totalDuration;
-  // no cut is observed
-  for (let i = 0; i < noOverlay.length; i++) {
-    duration += noOverlay[i].duration;
+function computeSelfTime(span: Span, allSpans: Span[]): number {
+  if (span.hasChildren) {
```

Comment:
nit: avoid extra nesting
```suggestion
  if (!span.hasChildren) {
      return span.duration;
  }
```